rest-service/manager_rest/test/endpoints/test_plugins_updates.py

```diff
@@ -0,0 +1,365 @@
+        self.client.plugins_update.update_plugins('hello_world')
+        self.client.plugins_update.update_plugins('hello_world', True)
+
+    @suppress_plugins_change_detection
+    def test_raises_with_real_active_plugins_updates_and_forced(self):
+        self.put_file(*self.put_blueprint_args(blueprint_id='hello_world'))
+        self.client.deployments.create('hello_world', 'd123')
+        self.wait_for_deployment_creation(self.client, 'd123')
+        plugins_update = self.client.plugins_update.update_plugins(
+            'hello_world')
+        execution = self._sm.get(models.Execution, plugins_update.execution_id)
+        execution.status = ExecutionState.STARTED
+        self._sm.update(execution)
+        with self.assertRaisesRegexp(
+                CloudifyClientError,
+                'There are plugins updates still active; the "force" flag '
+                'was used yet these updates have actual executions running '
+                'update IDs: {0}'.format(plugins_update.id)):
+            self.client.plugins_update.update_plugins('hello_world', True)
+
+    def test_raises_when_no_plugin_change_detected(self):
+        self.put_file(*self.put_blueprint_args(blueprint_id='hello_world'))
+        with self.assertRaisesRegexp(
+                CloudifyClientError,
+                'No new plugins changes have been detected when re-parsed '
+                'blueprint hello_world, aborting update.'):
+            self.client.plugins_update.update_plugins('hello_world')
+
+    @suppress_plugins_change_detection
+    def test_finalize_raises_plugins_update_with_not_compatible_state(self):
+        self.put_file(*self.put_blueprint_args(blueprint_id='hello_world'))
+        self.client.deployments.create('hello_world', 'd123')
+        self.wait_for_deployment_creation(self.client, 'd123')
+        plugins_update_id = self.client.plugins_update.update_plugins(
+            'hello_world').id
+        plugins_update = self._sm.get(models.PluginsUpdate, plugins_update_id)
+        plugins_update.state = STATES.SUCCESSFUL
+        self._sm.update(plugins_update)
+        with self.assertRaisesRegexp(
+                CloudifyClientError,
+                "Cannot finalize plugins update .+, it's not in the "
+                "{0} state\\.".format(STATES.EXECUTING_WORKFLOW)):
+            self.client.plugins_update.finalize_plugins_update(
+                plugins_update_id)
+
+    @suppress_plugins_change_detection
+    def test_finalize_updates_original_blueprint_plan(self):
+        self.put_file(*self.put_blueprint_args(blueprint_id='hello_world'))
+        self.client.deployments.create('hello_world', 'd123')
+        self.wait_for_deployment_creation(self.client, 'd123')
+        plugins_update_id = self.client.plugins_update.update_plugins(
+            'hello_world').id
+        blueprint = self._sm.get(models.Blueprint, 'hello_world')
+        plan = deepcopy(blueprint.plan)
+        plan.inputs['dummy'] = 'hi'
```

Comment:
why not check that the plugins are different? if the user did a plugin update only that part will change not the inputs

Reply:
This just checks that the plan has been updated, it doesn't really matter. For the sake of future implementations I made a change in the plugins key.